draft-irtf-cfrg-hpke.md

```diff
@@ -389,21 +420,21 @@ underlying the KDF.
 
 The GenerateKeyPair, Marshal, and Unmarshal functions are the same
 as for the underlying DH group.  The Marshal functions for the
-curves referenced in {{ciphersuites}} are as follows:
+groups referenced in {{ciphersuites}} are as follows:
 
 * P-256: A single byte set to 4, followed by the X-coordinate and the
   Y-coordinate of the point, encoded as 32-byte big-endian integers
 * P-384: A single byte set to 4, followed by the X-coordinate and the
   Y-coordinate of the point, encoded as 48-byte big-endian integers
 * P-521: A single byte set to 4, followed by the X-coordinate and the
   Y-coordinate of the point, encoded as 66-byte big-endian integers
-* Curve25519: The standard 32-byte representation of the public key
-* Curve448: The standard 56-byte representation of the public key
+* X25519: The standard 32-byte representation of the public key
```

Comment:
I really don't care strongly about this, but it seems like in RFC 7748, `X*` is the group, and `Curve*` is the ECDH operation (largely just specifying the base point).  Since we're doing ECDH here, `Curve*` seems like the right answer.

Reply:
I think you've got it backwards, though both specify a base point. X* is a function from bytes to bytes, with group operations and serialization built in. curve* is a curve with base point.

https://tools.ietf.org/html/rfc7748#section-4.1
>   For the ~128-bit security level, the prime 2^255 - 19 is recommended
>   for performance on a wide range of architectures.  Few primes of the
>   form 2^c-s with s small exist between 2^250 and 2^521, and other
>   choices of coefficient are not as competitive in performance.  This
>   prime is congruent to 1 mod 4, and the derivation procedure in
>   Appendix A results in the following Montgomery curve
>   v^2 = u^3 + A*u^2 + u, called "curve25519":

https://tools.ietf.org/html/rfc7748#section-5
>   The "X25519" and "X448" functions perform scalar multiplication on
>   the Montgomery form of the above curves.  (This is used when
>   implementing Diffie-Hellman.)  The functions take a scalar and a
>   u-coordinate as inputs and produce a u-coordinate as output.
>   Although the functions work internally with integers, the inputs and
>   outputs are 32-byte strings (for X25519) or 56-byte strings (for
>   X448) and this specification defines their encoding.

Reply:
Also https://mailarchive.ietf.org/arch/msg/cfrg/-9LEdnzVrE5RORux3Oo_oDDRksU/

Reply:
Yeah, I think using x25519 is also more clear since there are small subgroups of curve25519.

Reply:
You guys are correct.  I was looking at the subheadings of Section 6 (ECDH), when I should have been looking at Section 4.  Comment resolved.

Reply:
Hah. Those subheadings should probably have been X*. Ah well. :-)